### ISSUE TITLE
ypbind service failure on new CentOS 7 SLURM cluster

### ISSUE THREAD

**Issue by verdurin** [gc3-uzh-ch/elasticluster#402]
```
failed: [compute002] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute001] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute003] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute004] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute005] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute006] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute007] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute010] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute008] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
failed: [compute009] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}
```

**Comment by riccardomurri** [comment]
> failed: [compute002] (item=ypbind) => {"failed": true, "item": "ypbind", "msg": "Unable to start service ypbind: Job for ypbind.service failed because the control process exited with error code. See \"systemctl status ypbind.service\" and \"journalctl -xe\" for details.\n"}

Can you please post the output of `systemctl status ypbind` and
`tail -n 50 /var/log/messages` ?

Is this an existing cluster that got reconfigured, or a new one?

**Comment by verdurin** [comment]
```
[centos@frontend001 ~]$ systemctl status ypbind
● ypbind.service - NIS/YP (Network Information Service) Clients to NIS Domain Binder
   Loaded: loaded (/usr/lib/systemd/system/ypbind.service; disabled; vendor preset: disabled)
   Active: inactive (dead)
```

**Comment by verdurin** [comment]
```
[centos@frontend001 ~]$ sudo tail -n 50 /var/log/messages
Feb 28 17:58:33 frontend001 ansible-file: Invoked with directory_mode=None force=False remote_src=None path=/etc/munge/munge.key owner=munge follow=False group=munge unsafe_writes=None state=None content=NOT_LOGGING_PARAMETER serole=None diff_peek=None setype=None dest=/etc/munge/munge.key selevel=None original_basename=munge.key regexp=None validate=None src=munge.key seuser=None recurse=False delimiter=None mode=0400 backup=None
Feb 28 17:58:33 frontend001 ansible-systemd: Invoked with name=munge enabled=True daemon_reload=False state=started user=False masked=None
Feb 28 17:58:34 frontend001 ansible-stat: Invoked with checksum_algorithm=sha1 mime=False get_checksum=True path=/etc/slurm/slurm.conf checksum_algo=sha1 follow=False get_md5=False
Feb 28 17:58:35 frontend001 ansible-file: Invoked with directory_mode=None force=False remote_src=None path=/etc/slurm/slurm.conf owner=root follow=True group=root unsafe_writes=None serole=None content=NOT_LOGGING_PARAMETER state=None diff_peek=None setype=None dest=/etc/slurm/slurm.conf selevel=None original_basename=slurm.conf.j2 regexp=None validate=None src=None seuser=None recurse=False delimiter=None mode=0444 backup=None
Feb 28 17:58:35 frontend001 ansible-yum: Invoked with name=['openmpi-devel'] list=None disable_gpg_check=False conf_file=None install_repoquery=True state=present disablerepo=None update_cache=False enablerepo=None exclude=None validate_certs=True
Feb 28 17:58:35 frontend001 ansible-selinux: Invoked with policy=targeted configfile=/etc/selinux/config state=enforcing
Feb 28 17:58:35 frontend001 dbus-daemon: dbus[596]: avc:  received setenforce notice (enforcing=1)
Feb 28 17:58:35 frontend001 dbus[596]: avc:  received setenforce notice (enforcing=1)
Feb 28 17:58:35 frontend001 dbus[596]: [system] Reloaded configuration
Feb 28 17:58:35 frontend001 dbus-daemon: dbus[596]: [system] Reloaded configuration
Feb 28 17:58:36 frontend001 ansible-command: Invoked with warn=True executable=None _uses_shell=False _raw_params=restorecon -r / removes=None creates=None chdir=None
Feb 28 17:58:46 frontend001 ansible-yum: Invoked with name=['slurm', 'slurm-devel', 'slurm-perlapi', 'slurm-sjobexit', 'slurm-sjstat', 'slurm-torque'] list=None disable_gpg_check=False conf_file=None install_repoquery=True state=present disablerepo=None update_cache=False enablerepo=None exclude=None validate_certs=True
Feb 28 17:58:46 frontend001 ansible-systemd: Invoked with name=slurmd enabled=False daemon_reload=False state=stopped user=False masked=None
Feb 28 17:58:46 frontend001 ansible-systemd: Invoked with name=slurmctld enabled=False daemon_reload=False state=stopped user=False masked=None
Feb 28 17:58:46 frontend001 systemd: Reloading.
Feb 28 17:58:46 frontend001 systemd: Configuration file /usr/lib/systemd/system/auditd.service is marked world-inaccessible. This has no effect as configuration data is accessible via APIs without restrictions. Proceeding anyway.
Feb 28 17:58:47 frontend001 systemd: Stopping Slurm controller daemon...
Feb 28 17:58:47 frontend001 systemd: Stopped Slurm controller daemon.
Feb 28 17:58:47 frontend001 ansible-yum: Invoked with name=['mariadb-server', 'MySQL-python'] list=None disable_gpg_check=False conf_file=None install_repoquery=True state=latest disablerepo=None update_cache=False enablerepo=None exclude=None validate_certs=True
Feb 28 17:58:50 frontend001 ansible-systemd: Invoked with name=mariadb enabled=True daemon_reload=False state=started user=False masked=None
Feb 28 17:58:50 frontend001 ansible-mysql_db: Invoked with login_port=3306 ssl_cert=None config_file=/root/.my.cnf name=slurm encoding= login_user=None login_host=localhost ssl_key=None ssl_ca=None collation= login_unix_socket=None state=present login_password=NOT_LOGGING_PARAMETER quick=True single_transaction=False connect_timeout=30 target=None
Feb 28 17:58:51 frontend001 ansible-mysql_user: Invoked with ssl_key=None update_password=always login_user=None host_all=False priv=slurm.*:ALL state=present sql_log_bin=True encrypted=False connect_timeout=30 config_file=/root/.my.cnf login_host=localhost append_privs=False login_unix_socket=None host=frontend001 user=slurm login_password=NOT_LOGGING_PARAMETER check_implicit_admin=False password=NOT_LOGGING_PARAMETER login_port=3306 ssl_cert=None name=slurm ssl_ca=None
Feb 28 17:58:51 frontend001 ansible-mysql_user: Invoked with ssl_key=None update_password=always login_user=None host_all=False priv=slurm.*:ALL state=present sql_log_bin=True encrypted=False connect_timeout=30 config_file=/root/.my.cnf login_host=localhost append_privs=False login_unix_socket=None host=localhost user=slurm login_password=NOT_LOGGING_PARAMETER check_implicit_admin=False password=NOT_LOGGING_PARAMETER login_port=3306 ssl_cert=None name=slurm ssl_ca=None
Feb 28 17:58:51 frontend001 ansible-stat: Invoked with checksum_algorithm=sha1 mime=False get_checksum=True path=/etc/slurm/slurmdbd.conf checksum_algo=sha1 follow=False get_md5=False
Feb 28 17:58:51 frontend001 ansible-file: Invoked with directory_mode=None force=False remote_src=None path=/etc/slurm/slurmdbd.conf owner=root follow=True group=root unsafe_writes=None serole=None content=NOT_LOGGING_PARAMETER state=None diff_peek=None setype=None dest=/etc/slurm/slurmdbd.conf selevel=None original_basename=slurmdbd.conf.j2 regexp=None validate=None src=None seuser=None recurse=False delimiter=None mode=0444 backup=None
Feb 28 17:58:52 frontend001 ansible-yum: Invoked with name=['slurm-sql', 'slurm-slurmdbd'] list=None disable_gpg_check=False conf_file=None install_repoquery=True state=latest disablerepo=None update_cache=False enablerepo=None exclude=None validate_certs=True
Feb 28 17:58:54 frontend001 ansible-systemd: Invoked with name=slurmdbd enabled=True daemon_reload=False state=started user=False masked=None
Feb 28 17:58:55 frontend001 ansible-yum: Invoked with name=['mailx', 'slurm-plugins', 'slurm-slurmdb-direct', 'slurm'] list=None disable_gpg_check=False conf_file=None install_repoquery=True state=latest disablerepo=None update_cache=False enablerepo=None exclude=None validate_certs=True
Feb 28 17:58:57 frontend001 ansible-command: Invoked with warn=True executable=None _uses_shell=True _raw_params=sacctmgr --parsable --noheader list cluster | grep '^elasticluster|' || sacctmgr -i -Q add cluster elasticluster removes=None creates=None chdir=None
Feb 28 17:58:58 frontend001 ansible-command: Invoked with warn=True executable=None _uses_shell=True _raw_params=sacctmgr --immediate --parsable --noheader list account Cluster=elasticluster | grep '^root|' || sacctmgr -i --quiet add account root Cluster=elasticluster removes=None creates=None chdir=None
Feb 28 17:58:58 frontend001 ansible-command: Invoked with warn=True executable=None _uses_shell=True _raw_params=sacctmgr --immediate --parsable --noheader list user Account=root | grep '^centos|' || sacctmgr -i --quiet add user 'centos' DefaultAccount=root removes=None creates=None chdir=None
Feb 28 17:58:59 frontend001 ansible-systemd: Invoked with name=slurmctld enabled=True daemon_reload=False state=started user=False masked=None
Feb 28 17:58:59 frontend001 systemd: Reloading.
Feb 28 17:58:59 frontend001 systemd: Configuration file /usr/lib/systemd/system/auditd.service is marked world-inaccessible. This has no effect as configuration data is accessible via APIs without restrictions. Proceeding anyway.
Feb 28 17:58:59 frontend001 systemd: Starting Slurm controller daemon...
Feb 28 17:58:59 frontend001 systemd: Started Slurm controller daemon.
Feb 28 17:59:21 frontend001 ntpd[4859]: 0.0.0.0 0612 02 freq_set kernel -2.251 PPM
Feb 28 17:59:21 frontend001 ntpd[4859]: 0.0.0.0 0615 05 clock_sync
Feb 28 17:59:59 frontend001 systemd-logind: Removed session 4.
Feb 28 17:59:59 frontend001 systemd: Removed slice user-1000.slice.
Feb 28 17:59:59 frontend001 systemd: Stopping user-1000.slice.
Feb 28 18:01:01 frontend001 systemd: Created slice user-0.slice.
Feb 28 18:01:01 frontend001 systemd: Starting user-0.slice.
Feb 28 18:01:01 frontend001 systemd: Started Session 5 of user root.
Feb 28 18:01:01 frontend001 systemd: Starting Session 5 of user root.
Feb 28 18:21:34 frontend001 systemd: Created slice user-1000.slice.
Feb 28 18:21:34 frontend001 systemd: Starting user-1000.slice.
Feb 28 18:21:34 frontend001 systemd-logind: New session 6 of user centos.
Feb 28 18:21:34 frontend001 systemd: Started Session 6 of user centos.
Feb 28 18:21:34 frontend001 systemd: Starting Session 6 of user centos.
```

**Comment by verdurin** [comment]
This is a new cluster, created immediately after the new installation via virtualenv.

**Comment by riccardomurri** [comment]
I cannot reproduce this on my site's OpenStack, using CentOS 7.2.1511
(official CentOS7 cloud image)...

`systemctl status` output is not particularly informative and the logs
do not mention ypbind :-( -- did you get them from the front-end or from
a compute node?

Can you try running the following commands on a compute node and post
the output?  (leave the last ypbind running for at least 1 minute)

    rpcinfo -p localhost

    rpcinfo -p master001

    ypbind -no-dbus -debug

**Comment by verdurin** [comment]
Actually, I can't SSH to the nodes from the frontend, as I think I've reported elsewhere, at least not without further intervention.

**Comment by verdurin** [comment]
After connecting manually (i.e. directly from the elasticluster host, not from the frontend):

```
[centos@compute001 ~]$ rpcinfo -p localhost
   program vers proto   port  service
    100000    4   tcp    111  portmapper
    100000    3   tcp    111  portmapper
    100000    2   tcp    111  portmapper
    100000    4   udp    111  portmapper
    100000    3   udp    111  portmapper
    100000    2   udp    111  portmapper
```
(Assumed you meant frontend001 - there's no master001 in this cluster)
```
[centos@compute001 ~]$ rpcinfo -p frontend001
   program vers proto   port  service
    100000    4   tcp    111  portmapper
    100000    3   tcp    111  portmapper
    100000    2   tcp    111  portmapper
    100000    4   udp    111  portmapper
    100000    3   udp    111  portmapper
    100000    2   udp    111  portmapper
    100009    1   udp    779  yppasswdd
    100004    2   udp    835  ypserv
    100004    1   udp    835  ypserv
    100004    2   tcp    838  ypserv
    100004    1   tcp    838  ypserv
```

```
[centos@compute001 ~]$ sudo systemctl status ypbind
● ypbind.service - NIS/YP (Network Information Service) Clients to NIS Domain Binder
   Loaded: loaded (/usr/lib/systemd/system/ypbind.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Tue 2017-02-28 17:59:53 UTC; 6h ago
  Process: 8260 ExecStartPost=/usr/libexec/ypbind-post-waitbind (code=exited, status=1/FAILURE)
  Process: 8255 ExecStart=/usr/sbin/ypbind -n $OTHER_YPBIND_OPTS (code=exited, status=0/SUCCESS)
  Process: 8252 ExecStartPre=/usr/sbin/setsebool allow_ypbind=1 (code=exited, status=0/SUCCESS)
  Process: 8248 ExecStartPre=/usr/libexec/ypbind-pre-setdomain (code=exited, status=0/SUCCESS)
 Main PID: 8255 (code=exited, status=0/SUCCESS)
   Status: "Processing requests..."

Feb 28 17:59:05 compute001 systemd[1]: Starting NIS/YP (Network Information Service) Clients to NIS Domain Binder...
Feb 28 17:59:05 compute001 setsebool[8252]: The allow_ypbind policy boolean was changed to 1 by root
Feb 28 17:59:53 compute001 systemd[1]: ypbind.service: control process exited, code=exited status=1
Feb 28 17:59:53 compute001 systemd[1]: Failed to start NIS/YP (Network Information Service) Clients to NIS Domain Binder.
Feb 28 17:59:53 compute001 systemd[1]: Unit ypbind.service entered failed state.
Feb 28 17:59:53 compute001 systemd[1]: ypbind.service failed.
```

```
[centos@compute001 ~]$ sudo ypbind -no-dbus -debug
18576: parsing config file
18576: Trying entry: domain elasticluster server frontend001
18576: parsed domain 'elasticluster' server 'frontend001'
18576: add_server() domain: elasticluster, host: frontend001, slot: 0
18576: [Welcome to ypbind-mt, version 1.37.1]

18576: ping interval is 20 seconds

18576: rebind interval is 900 seconds

18578: ping host 'frontend001', domain 'elasticluster'
18578: host 'frontend001' doesn't answer.
18578: Pinging all active servers.
18578: ping host 'frontend001', domain 'elasticluster'
18578: host 'frontend001' doesn't answer.
18578: Pinging all active servers.
18578: ping host 'frontend001', domain 'elasticluster'
18578: host 'frontend001' doesn't answer.
18578: Pinging all active servers.
18578: ping host 'frontend001', domain 'elasticluster'
18578: host 'frontend001' doesn't answer.
```

From `/var/log/messages`:

```
Feb 28 17:59:05 extra-network-test-6 ansible-systemd: Invoked with name=ypbind enabled=True daemon_reload=False state=started user=False masked=None
Feb 28 17:59:05 extra-network-test-6 systemd: Starting NIS/YP (Network Information Service) Clients to NIS Domain Binder...
Feb 28 17:59:05 extra-network-test-6 setsebool: The allow_ypbind policy boolean was changed to 1 by root
Feb 28 17:59:05 extra-network-test-6 dbus[594]: avc:  received policyload notice (seqno=3)
Feb 28 17:59:05 extra-network-test-6 dbus-daemon: dbus[594]: avc:  received policyload notice (seqno=3)
Feb 28 17:59:05 extra-network-test-6 dbus[594]: [system] Reloaded configuration
Feb 28 17:59:05 extra-network-test-6 dbus-daemon: dbus[594]: [system] Reloaded configuration
Feb 28 17:59:05 extra-network-test-6 ypbind: Binding NIS service
Feb 28 17:59:53 extra-network-test-6 ypbind: Binding took 48 seconds
Feb 28 17:59:53 extra-network-test-6 ypbind: NIS server for domain elasticluster is not responding.
Feb 28 17:59:53 extra-network-test-6 ypbind: Killing ypbind with PID 8255.
Feb 28 17:59:53 extra-network-test-6 ypbind: Try increase NISTIMEOUT in /etc/sysconfig/ypbind
Feb 28 17:59:53 extra-network-test-6 systemd: ypbind.service: control process exited, code=exited status=1
Feb 28 17:59:53 extra-network-test-6 systemd: Failed to start NIS/YP (Network Information Service) Clients to NIS Domain Binder.
Feb 28 17:59:53 extra-network-test-6 systemd: Unit ypbind.service entered failed state.
Feb 28 17:59:53 extra-network-test-6 systemd: ypbind.service failed.
```

**Comment by verdurin** [comment]
I started another new cluster and again setup failed at the same point.

**Comment by riccardomurri** [comment]
(Adam Huffman, Tue, Feb 28, 2017 at 04:05:45PM -0800:)
> Actually, I can't SSH to the nodes from the frontend, as I think I've reported elsewhere, at least not without further intervention.

I cannot reproduce this either :-(  Is there any chance I can get a copy
of the VM image/snapshot that you're using, to try it out locally?

**Comment by riccardomurri** [comment]
Thanks for the logs.

From this we it looks like both `rpcbind`/`portmap` and `ypserv` are
correctly running on the master/front-end node::

> ```
> [centos@compute001 ~]$ rpcinfo -p frontend001
>    program vers proto   port  service
> ...
>     100009    1   udp    779  yppasswdd
>     100004    2   udp    835  ypserv
>     100004    1   udp    835  ypserv
>     100004    2   tcp    838  ypserv
>     100004    1   tcp    838  ypserv
> ```

However, from `ypbind -debug` we see that it cannot connect to the
master::

> ```
> [centos@compute001 ~]$ sudo ypbind -no-dbus -debug
> ...
> 18578: ping host 'frontend001', domain 'elasticluster'
> 18578: host 'frontend001' doesn't answer.
> ```

My guess (given also that you cannot SSH from frontend to comnpute
nodes), would be: either a firewall issues (iptables rules that
ElastiCluster doesn't expect), or that the security group does not allow
inter-host connections.

Can you please check:

* what does `iptables -v -L` say on *both* frontend and compute nodes?
* what rules are in the security group(s) that the VM have been created into?

**Comment by verdurin** [comment]
The security group rules are:

```
+-------------+-----------+---------+-----------+--------------+
| IP Protocol | From Port | To Port | IP Range  | Source Group |
+-------------+-----------+---------+-----------+--------------+
| icmp        | -1        | -1      | 0.0.0.0/0 |              |
| tcp         | 22        | 22      | 0.0.0.0/0 |              |
| tcp         | 1         | 65535   |           | vanloo       |
+-------------+-----------+---------+-----------+--------------+
```

**Comment by verdurin** [comment]
frontend iptables:

```
[centos@frontend001 ~]$ sudo iptables -v -L
Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
81689   57M ACCEPT     all  --  any    any     anywhere             anywhere             state RELATED,ESTABLISHED
   63  4876 ACCEPT     all  --  lo     any     anywhere             anywhere
    0     0 ACCEPT     all  --  any    any     frontend001          anywhere
    0     0 ACCEPT     all  --  any    any     compute001           anywhere
    0     0 ACCEPT     all  --  any    any     compute002           anywhere
    0     0 ACCEPT     all  --  any    any     compute003           anywhere
    0     0 ACCEPT     all  --  any    any     compute004           anywhere
    0     0 ACCEPT     all  --  any    any     compute005           anywhere
    0     0 ACCEPT     all  --  any    any     compute006           anywhere
    0     0 ACCEPT     all  --  any    any     compute007           anywhere
    0     0 ACCEPT     all  --  any    any     compute008           anywhere
    0     0 ACCEPT     all  --  any    any     compute009           anywhere
    0     0 ACCEPT     all  --  any    any     compute010           anywhere
    3   164 ACCEPT     tcp  --  any    any     anywhere             anywhere             tcp dpt:ssh

Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 79858 packets, 4914K bytes)
 pkts bytes target     prot opt in     out     source               destination
```

**Comment by verdurin** [comment]
compute iptables:

```
[centos@compute010 ~]$ sudo iptables -v -L
Chain INPUT (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
 2140 1327K ACCEPT     all  --  any    any     anywhere             anywhere             state RELATED,ESTABLISHED
   37  2880 ACCEPT     all  --  lo     any     anywhere             anywhere
 5231  314K ACCEPT     all  --  any    any     frontend001          anywhere
    0     0 ACCEPT     all  --  any    any     compute001           anywhere
    0     0 ACCEPT     all  --  any    any     compute002           anywhere
   12   912 ACCEPT     all  --  any    any     compute003           anywhere
    0     0 ACCEPT     all  --  any    any     compute004           anywhere
    0     0 ACCEPT     all  --  any    any     compute005           anywhere
    0     0 ACCEPT     all  --  any    any     compute006           anywhere
    0     0 ACCEPT     all  --  any    any     compute007           anywhere
    0     0 ACCEPT     all  --  any    any     compute008           anywhere
    0     0 ACCEPT     all  --  any    any     compute009           anywhere
    0     0 ACCEPT     all  --  any    any     compute010           anywhere
    4   224 ACCEPT     tcp  --  any    any     anywhere             anywhere             tcp dpt:ssh

Chain FORWARD (policy DROP 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 8616 packets, 496K bytes)
 pkts bytes target     prot opt in     out     source               destination
```

**Comment by riccardomurri** [comment]
> The security group rules are:
> 
> ```
> +-------------+-----------+---------+-----------+--------------+
> | IP Protocol | From Port | To Port | IP Range  | Source Group |
> +-------------+-----------+---------+-----------+--------------+
> | icmp        | -1        | -1      | 0.0.0.0/0 |              |
> | tcp         | 22        | 22      | 0.0.0.0/0 |              |
> | tcp         | 1         | 65535   |           | vanloo       |
> +-------------+-----------+---------+-----------+--------------+
> ```

...and are the VMs of the cluster created in group `vanloo`?

Can you please test connectivity of compute nodes to front-end?

    nc -v frontend001 835 < /dev/null
    nc -v frontend001 838 < /dev/null

Conversely, from frontend to compute::

    nc -v compute001 111 < /dev/null

(You might need to run `yum install nc` first.)

**Comment by riccardomurri** [comment]
Can you change the rules of the security group and allow also UDP from
all hosts in the `vanloo` security group?

**Comment by verdurin** [comment]
I've added UDP to the security group, and now it's failing at:

```
TASK [nfs-client : add to /etc/fstab] ******************************************
task path: /home/centos/src/venv-2017-02-28/elasticluster/src/elasticluster/share/playbooks/roles/nfs-client/tasks/nfsmount.yml:8
fatal: [compute009]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute008]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute006]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute010]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute007]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute005]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute002]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute004]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute001]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
fatal: [compute003]: FAILED! => {"changed": false, "failed": true, "msg": "Error mounting /home: mount.nfs: requested NFS version or transport protocol is not supported\n"}
```

**Comment by verdurin** [comment]
```
[centos@compute010 ~]$ sudo mount -v frontend001:/home /home
mount.nfs: timeout set for Wed Mar  1 16:51:10 2017
mount.nfs: trying text-based options 'vers=4,addr=10.2.210.84,clientaddr=10.2.210.90'
mount.nfs: mount(2): Connection refused
mount.nfs: trying text-based options 'addr=10.2.210.84'
mount.nfs: prog 100003, trying vers=3, prot=6
mount.nfs: portmap query retrying: RPC: Program not registered
mount.nfs: prog 100003, trying vers=3, prot=17
mount.nfs: portmap query failed: RPC: Program not registered
mount.nfs: requested NFS version or transport protocol is not supported
```

**Comment by verdurin** [comment]
Hmm.

The NFS service wasn't running...

**Comment by riccardomurri** [comment]
(Adam Huffman, Wed, Mar 01, 2017 at 08:40:27AM -0800:)
> I've added UDP to the security group, and now it's failing at: `TASK: [nfs-client : add to /etc/fstab]`

OK, so the problem with NIS/YP is solved, right?  And the solution is that
the security group should allow unrestricted UDP.

**Comment by riccardomurri** [comment]
> fatal: [compute009]: FAILED! => {"changed": false, "failed": true,
> "msg": "Error mounting /home: mount.nfs: requested NFS version or
> transport protocol is not supported\n"}

This I can reproduce.  However, can you please open a different issue
for it?  I'd like issue reports to be focused on one problem only,
otherwise they become unreadable.

(The problem is that somehow the task to install
`nfs-utils` (which provides the `nfs` service) is skipped.  Short-term
workaround is to log in to the frontend node and run `yum install
nfs-utils`.  Unfortunately I  won't have time to patch the code until
late next week.)

**Comment by verdurin** [comment]
Yes, the original ypbind problem was solved by the security group change.

**Comment by verdurin** [comment]
Sure, I'll report that separately.

Thanks very much for the swift assistance. Is it worth sending you a PR about the security group requirement?

**Comment by riccardomurri** [comment]
> Is it worth sending you a PR about the security group requirement?

Yes, please!

**Comment by verdurin** [comment]
Closing - this was resolved by allowing all UDP traffic within instances using the same security group.

**Comment by HackerMD** [comment]
I am also getting this error. I double-checked security groups - ingressing UDP connections are allowed from all nodes within the cluster.

`journalctl` releaved the following:

```
May 28 20:54:47 performance-slurm-frontend-001 ansible-systemd[12373]: Invoked with no_block=False name=rpcbind enabled=None daemon_reload=False state=restarted user=False masked=None
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Stopping NIS/YP (Network Information Service) Users Passwords Change Server...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Stopping NIS/YP (Network Information Service) Server...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Stopping RPC bind service...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Starting RPC bind service...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Started RPC bind service.
May 28 20:54:47 performance-slurm-frontend-001 sudo[12398]:   centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/sh -c echo BECOME-SUCCESS-xpscjlndpzqeilrpezfhtrqqptstvkal; /usr/bin/python
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Starting NIS/YP (Network Information Service) Users Passwords Change Server...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Starting NIS/YP (Network Information Service) Server...
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Started NIS/YP (Network Information Service) Server.
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Started NIS/YP (Network Information Service) Users Passwords Change Server.
May 28 20:54:47 performance-slurm-frontend-001 ansible-systemd[12405]: Invoked with no_block=False name=ypbind enabled=None daemon_reload=False state=restarted user=False masked=None
May 28 20:54:47 performance-slurm-frontend-001 systemd[1]: Starting NIS/YP (Network Information Service) Clients to NIS Domain Binder...
May 28 20:54:47 performance-slurm-frontend-001 dbus[546]: avc:  received policyload notice (seqno=2)
May 28 20:54:47 performance-slurm-frontend-001 dbus-daemon[546]: dbus[546]: avc:  received policyload notice (seqno=2)
May 28 20:54:47 performance-slurm-frontend-001 setsebool[12415]: The allow_ypbind policy boolean was changed to 1 by root
May 28 20:54:47 performance-slurm-frontend-001 dbus[546]: [system] Reloaded configuration
May 28 20:54:47 performance-slurm-frontend-001 dbus-daemon[546]: dbus[546]: [system] Reloaded configuration
May 28 20:54:47 performance-slurm-frontend-001 ypbind[12425]: Binding NIS service
May 28 20:55:33 performance-slurm-frontend-001 ypbind[12518]: Binding took 46 seconds
May 28 20:55:33 performance-slurm-frontend-001 ypbind[12520]: NIS server for domain elasticluster is not responding.
May 28 20:55:33 performance-slurm-frontend-001 ypbind[12521]: Killing ypbind with PID 12419.
May 28 20:55:33 performance-slurm-frontend-001 ypbind[12522]: Try increase NISTIMEOUT in /etc/sysconfig/ypbind
May 28 20:55:33 performance-slurm-frontend-001 systemd[1]: ypbind.service: control process exited, code=exited status=1
May 28 20:55:33 performance-slurm-frontend-001 systemd[1]: Failed to start NIS/YP (Network Information Service) Clients to NIS Domain Binder.
May 28 20:55:33 performance-slurm-frontend-001 systemd[1]: Unit ypbind.service entered failed state.
May 28 20:55:33 performance-slurm-frontend-001 systemd[1]: ypbind.service failed.
```

After this failure, I ran the playbook again and the second time it ran through without error.

**Comment by verdurin** [comment]
I saw exactly what @HackerMD reported on a new cluster today. It failed the first time, then running `setup` succeeded.

**Comment by pescobar** [comment]
I observer the same. The first run fails, when doing `setup` again it works. My security group allows all traffic in the internal network

```
(openstack) security group rule list compute-clusters
+--------------------------------------+-------------+-------------+------------+-----------------------+
| ID                                   | IP Protocol | IP Range    | Port Range | Remote Security Group |
+--------------------------------------+-------------+-------------+------------+-----------------------+
| 1b897120-09b1-46a1-acf8-20003d90fbf3 | tcp         | 10.0.0.0/16 | 1:65535    |                       |
| 2e31866d-e7f5-4ee7-89b1-d84b12e3c1f3 | icmp        | 10.0.0.0/16 |            |                       |
| 4a85bef1-82bc-4288-9a14-2aa81c1e447a | udp         | 10.0.0.0/16 | 1:65535    |                       |
+--------------------------------------+-------------+-------------+------------+-----------------------+
```

**Comment by pescobar** [comment]
btw, after `setup` completes correctly I had to ssh to the fronted node and do `sudo systemctl restart slurm` so slurm starts working

**Comment by riccardomurri** [comment]
@hackermd @pescobar @verdurin This additional problem of `ypbind` starting on the front-end node should be fixed in commit 6c8c8f8
